### PR TITLE
Enhancement/181 add max length in the name field

### PR DESCRIPTION
**Pull request by tanmaythole**

### CODE REVIEW

File: src/Components/ProjectsComponent/PayloadComponents/AddNewPayload.jsx [hunk 1]
```diff
@@ -23,7 +25,9 @@ const AddNewPayload = ({ cat, data, isLoading, onchange, handleSubmit }) => {
   };
 
   const onPayloadFieldsChange = (result) => {
-    onchange("payload", result);
+    if (JSON.parse(result)) {
+      onchange("payload", result);
```
Comment (ChinmaySonawane) [inline src/Components/ProjectsComponent/PayloadComponents/AddNewPayload.jsx]: `onChange`

File: src/Components/ProjectsComponent/EndpointComponent/AddNewEndpoint.jsx [hunk 2]
```diff
@@ -7,46 +7,51 @@ import { ViewComponent } from "Components/CustomComponents";
 
 import { convertToSlug } from "utils";
 
+import { NAME_FIELD_MAX_LENGTH } from "constants/appConstants";
+
 const AddNewEndpoint = ({ cat, data, isLoading, onchange, handleSubmit }) => {
```
Comment (ChinmaySonawane) [inline src/Components/ProjectsComponent/EndpointComponent/AddNewEndpoint.jsx]: `onChange`

File: src/Components/ProjectsComponent/EnvironmentComponents/AddNewEnvironment.jsx [hunk 3]
```diff
@@ -16,45 +18,48 @@ const AddNewEnvironment = ({
...
+        onSave={handleSubmit}
+        isLoading={isLoading}
+        isSaveDisabled={isSaveButtonDisabled}
+      >
+        <FormInput
+          label="Name"
+          placeholder="Name"
+          name="name"
+          value={name}
+          onChange={onchange}
```
Comment (ChinmaySonawane) [inline src/Components/ProjectsComponent/EnvironmentComponents/AddNewEnvironment.jsx]: `onChange`

File: src/Components/ProjectsComponent/HeaderComponents/AddNewHeader.jsx [hunk 4]
```diff
@@ -18,45 +20,48 @@ const AddNewHeader = ({
...
+        onSave={handleSubmit}
+        isLoading={isLoading}
+        isSaveDisabled={isSaveButtonDisabled}
+      >
+        <FormInput
+          label="Name"
+          placeholder="Name"
+          name="name"
+          value={name}
+          onChange={onchange}
```
Comment (ChinmaySonawane) [inline src/Components/ProjectsComponent/HeaderComponents/AddNewHeader.jsx]: `onChange`

File: src/Components/ProjectsComponent/PayloadComponents/AddNewPayload.jsx (hunk 1 above)
Comment (tanmaythole) [reply src/Components/ProjectsComponent/PayloadComponents/AddNewPayload.jsx]: 👍🏻

File: src/Components/ProjectsComponent/EnvironmentComponents/AddNewEnvironment.jsx (hunk 3 above)
Comment (tanmaythole) [reply src/Components/ProjectsComponent/EnvironmentComponents/AddNewEnvironment.jsx]: 👍🏻

File: src/Components/ProjectsComponent/HeaderComponents/AddNewHeader.jsx (hunk 4 above)
Comment (tanmaythole) [reply src/Components/ProjectsComponent/HeaderComponents/AddNewHeader.jsx]: 👍🏻

File: src/Components/ProjectsComponent/EndpointComponent/AddNewEndpoint.jsx (hunk 2 above)
Comment (tanmaythole) [reply src/Components/ProjectsComponent/EndpointComponent/AddNewEndpoint.jsx]: 👍🏻

**Review by ChinmaySonawane — APPROVED** [review]

**Pull request merged**